#265 live spectating player not working

DjFrytek opened the pull request:
Closes #265

DjFrytek committed:
Fixed camera not locking in spectator mode

Deleted setting spectatorMode to true in every update and then checking for it right after.

DjFrytek committed:
Fixed exception when switching to previous spectated

When spectated player index was below 0 it went to index equal to number of players, which is always 1 higher than maximal possible index, now it goes to number of players - 1.

DjFrytek committed:
Changed version number.

Changed version in VersionManager from to ISSUE-#265

Lithrun reviewed (changes requested):
While testing, I've noticed:
- When you go into specmode, it is now always required to spectate another player. You can no longer freely roam around within the specmode camera because of this. It is expected that pressing a key (In AoTTG this is right click, although we can change this to perhaps the WASD keys) will give the player control over the camera again.

DjFrytek committed:
Fixed exiting spectating player

Broken up alpha1 into two different keys. Alpha1 opens choose side window, AttackSpecial (Right mouse button), toggles specMode. Changed EnterSpecMode method, to not require choosing side when exiting spec mode, as that was causing an issue and wasn't supposed to happen.

DjFrytek committed:
Added instuction how to enable freeflight camera.

Added instruction in the middle of the screen, under the other control instructions explaining how to enable freeflight camera.

DjFrytek committed:
Fixed toggling spectator mode

The functions changing spectator mode used different methods to do so, because of which if spectator mode was changed by anything else than ToggleSpectatorMode it was getting bugged, I moved some code from ToggleSpectatorMode the method it and other functionalities used, so it functions properly no matter how the spectator mode was changed.

Lithrun reviewed (changes requested):
Code changes seem fine, but there's still a merge conflict. Resolve this, then will do a final review.

DjFrytek committed:
Merge branch 'development' into #265-Live-spectating-player-not-working

DjFrytek committed:
Replaced all the settings[0xf5] references in MainCamera to spectatorMode

Made spectatorMode static, made IsSpecmode return spectatorMode instead of settings.
Made setSpectatorMode use spectatorMode instead of settings, same goes to ToggleSpecMode.

DjFrytek committed:
Merge branch '#265-Live-spectating-player-not-working' of https://github.com/AoTTG-2/AoTTG-2 into #265-Live-spectating-player-not-working

DjFrytek committed:
setSpectatorMode meaning fixed

setSpectatorMode no longer need to be reversed depending what calls it.

DjFrytek committed:
Changed spectatorMode visibility

Changed spectatorMode and ToggleSpecMode() to non static, and made spectatorMode private, as Sonarcloud suggested.

DjFrytek committed:
Chaned toggle spectatorMode line to a simpler one

Lithrun commented:
Assigning myself to this issue to resolve the code smell, and resolve the merge conflict as per request of @DjFrytek

Lithrun committed:
Merge /w Development

Lithrun committed:
Merge /w development
